handler/src/main/java/io/netty/handler/ssl/SslHandler.java

```diff
@@ -538,18 +548,14 @@ public void flush(ChannelHandlerContext ctx) throws Exception {
         }
         try {
             wrap(ctx, false);
-        } catch (Throwable cause) {
-            // Fail pending writes.
-            pendingUnencryptedWrites.removeAndFailAll(cause);
```

Comment:
We should make a best effort to keep our state consistent and let higher layers determine if the condition is recoverable or not. If we want to re-throw then that is fine, but if we let control flow escape then all bets are off, and the upper layers have even less chance of recovering (without knowing this is the case). I agree it is unlikely to recover in the event of OOME but since this is part of the control flow it seems strange to ignore it. Its not the same but I try to draw an analogy to a non-memory managed environment:

```c
int* x = malloc(10 * sizeof(int));
// assume this worked.
int* y = malloc(10 * sizeof(int));
if (y == NULL) {
  // what would you do here?
  // generally you would free `x` and return some sort of error condition.
  // isn't this similar to an OOME in a non-memory managed environment?
}
```